### PR TITLE
Ice balance attempt 3

### PR DESCRIPTION
**Pull request by mikepmanning**
All stuff for ice curing

Adds new affs and replaces necessary affs with overhaul stuff.

### CODE REVIEW

**Comment by vadi2** [comment]
Looks like there's a merge conflict you need to resolve in the branch first.

**Comment by mikepmanning** [comment]
How do I see what the conflicts are?

**Comment by vadi2** [comment]
I use SmartGit, so I clicked on the "command line instructions" - and it's Github giving command line instructions for merging the Ice-Balance-attempt-3 into your master branch. So I did that using the SmartGit UI instead, and there was a conflict where both branches modified some of the same lines in defs.lua. I set the changes from Ice-Balance-attempt-3 to override them.

Going to see if I can submit a pull request to your fork with the merge conflicts resolved that should update then your original pull request.

**Comment by vadi2** [comment]
Okay it's https://github.com/siddister/m-mf/pull/3, go and approve that one.

You modified the waterbreathing lines and submitted a PR for that into m-mf/master and then modified them separately in the ice balance branch without pulling in your changes from master - so you've made 2 separate edits to the same stuff, hence the conflict.

**Comment by mikepmanning** [comment]
Alright done, my bad there.

**Comment by vadi2** [comment]
All good. Some fixes needed -

^The salve mends your (head|chest|gut|left arm|left leg|right arm|right leg), leaving it with (light|heavy|critical) wounds.$

add a \ before a real . at the end on a few regex triggers.

**Comment by vadi2** [comment]
<regex>^mmsp convert$</regex>

Toggles should also provide a way of not toggling but explicitly setting an option, look at one of the current toggles for an example on how to do that.

**Comment by vadi2** [comment]
Remember to pull your branch before you make changes so you get my merge conflict resolution in.

File: raw-mm.aliases.lua [hunk 1]
```diff
@@ -458,13 +458,8 @@ function ashow()
   echoLink("steam", 'mm.printorder("steam")', 'View steam priorities', true)
   setUnderline(false) echo", " setUnderline(true)
   echoLink("wafer", 'mm.printorder("wafer")', 'View wafer priorities', true)
-  resetFormat()
-  echo"\n"
-
-  echofn("Arena mode:       ")
-  setFgColor(unpack(getDefaultColorNums))
-  setUnderline(true)
-  echoLink(conf.arena and "enabled" or "disabled", "$(sys).tntf_set('arena', "..(conf.arena and "false" or "true").. ', false); svo.ashow()', (conf.arena and "Disable" or "Enable")..' arena triggers', true)
```
Comment (vadi2) [inline raw-mm.aliases.lua:467]: How come you got rid of showing arena mode?

**Comment by vadi2** [comment]
Any reason valid.oldwarrior isn't a conf option?

**Comment by vadi2** [comment]
In bruising.isadvisable, you might want to be checking doingaction("bleeding") as well... otherwise I think both bleeding and bruising will be sending clot commands and it could snowball.

**Comment by vadi2** [comment]
The old warrior toggle isn't used everywhere, ie:

```
  oncompleted = function ()
    removeaff("collapsedlungs")
    --removeaff("paregenchest")
    --addaff(dict.puncturedlung)
  end,
```

**Comment by vadi2** [comment]
if $(sys)["version"] >= 4 then looks dodgy... did you mean to use the oldwarrior check?

File: raw-mm.sp.lua [hunk 2]
```diff
@@ -123,16 +145,28 @@ sps.install = {
       local function makecodestrings(name)
         local t = {}
         t[#t+1] = 'mm.sp.setstancelevel("'..name..'", false, true)'
-        for amount = 275, 2000, 275 do
-          t[#t+1] = 'mm.sp.setstancelevel("'..name..'", '..amount..', true)'
+        if $(sys)["version"] >= 4 then
```
Comment (vadi2) [inline raw-mm.sp.lua:148]: Looks dodgy... did you mean to use the oldwarrior check?

**Comment by mikepmanning** [comment]
I'll take care of all this when I get home - most of it should be easy. I was rushing to get things done so people would have working systems with the release

**Comment by vadi2** [comment]
Okay do the fixes in another PR, I'll merge and release this.

**Comment by vadi2** [comment]
Okay I've created the release (https://github.com/m-mf/m-mf/releases/tag/4) - no time to add a detailed changelog for all the stuff. Could you write a changelog I should enter and make all the announcements?

**Comment by mikepmanning** [comment]
Yeah, want me to post to forums too?

**Comment by vadi2** [comment]
Yeah - forums, m&m clan, etc.

Mention the new toggle alias you've got as well.

**Comment by mikepmanning** [comment]
Changelog - version 4
- fixed generic def lines (mercy, kingdom, beauty, perfection, acquisitio, waterbreathing) 
- added ice balance and overhaul afflictions associated with warrior overhaul
- added bruising
- added stance/parry conversion
- combatstyle defs have been added for all the specs
- added 'mmsp convert' alias to switch between old warrior system and new warrior system in case of reversion
- added in overhaul cures for anorexia, slickness and asthma
- added in overhaul damagedlimbs and mutilatedlimbs to replace crippledlimbs and mangled/missinglimbs
- added damagedskull to replace fracturedskull with overhaul
- adjusted lucidity to work with anorexia so it can cure it
- damagedthroat blocks lucidity (may need to have it block other sipping)

**Comment by vadi2** [comment]
Changelog updated.

**Comment by vadi2** [comment]
I don't think you've added default priorities for ice by the way - the
bin/default_prios file doesn't list them.

**Pull request merged**